From ce66990efe9a73176c4a6f6ee4b40590cbf03aaf Mon Sep 17 00:00:00 2001
From: "Peter A. Bigot" <pab@pabigot.com>
Date: Tue, 27 Aug 2013 05:41:17 -0500
Subject: [PATCH 7/7] OMAP3: overo: add Alto35 expansion board support

Pin configuration per branched commit b96d959acd80:

  "Tweak pin mux for Alto35.  GPIOs 148 and 151 used.  Disable SPI CS1 as
  the screen configuration information is incorrect"

Modified to apply only to Alto35 boards and to initialize all LEDs and
user-controllable button.

Upstream-Status: Pending

Signed-off-by: Peter A. Bigot <pab@pabigot.com>
---
 board/overo/overo.c |   16 ++++++++++++++--
 board/overo/overo.h |    8 ++++++++
 2 files changed, 22 insertions(+), 2 deletions(-)

diff --git a/board/overo/overo.c b/board/overo/overo.c
index c70fcc3..397aa70 100644
--- a/board/overo/overo.c
+++ b/board/overo/overo.c
@@ -56,6 +56,7 @@ DECLARE_GLOBAL_DATA_PTR;
 #define GUMSTIX_CHESTNUT43		0x06000200
 #define GUMSTIX_PINTO			0x07000200
 #define GUMSTIX_GALLOP43		0x08000200
+#define GUMSTIX_ALTO35			0x09000200
 
 #define ETTUS_USRP_E			0x01000300
 
@@ -247,6 +248,8 @@ unsigned int get_expansion_id(void)
  */
 int misc_init_r(void)
 {
+	unsigned int expansion_id;
+
 	twl4030_power_init();
 	twl4030_led_init(TWL4030_LED_LEDEN_LEDAON | TWL4030_LED_LEDEN_LEDBON);
 
@@ -269,7 +272,8 @@ int misc_init_r(void)
 		puts("Unable to detect mmc2 connection type\n");
 	}
 
-	switch (get_expansion_id()) {
+	expansion_id = get_expansion_id();
+	switch (expansion_id) {
 	case GUMSTIX_SUMMIT:
 		printf("Recognized Summit expansion board (rev %d %s)\n",
 			expansion_config.revision,
@@ -319,6 +323,13 @@ int misc_init_r(void)
 			expansion_config.fab_revision);
 		setenv("defaultdisplay", "lcd43");
 		break;
+	case GUMSTIX_ALTO35:
+		printf("Recognized Alto35 expansion board (rev %d %s)\n",
+			expansion_config.revision,
+			expansion_config.fab_revision);
+		MUX_ALTO35();
+		setenv("defaultdisplay", "lcd35");
+		break;
 	case ETTUS_USRP_E:
 		printf("Recognized Ettus Research USRP-E (rev %d %s)\n",
 			expansion_config.revision,
@@ -330,7 +341,8 @@ int misc_init_r(void)
 		puts("No EEPROM on expansion board\n");
 		break;
 	default:
-		puts("Unrecognized expansion board\n");
+		printf("Unrecognized expansion board 0x%08x\n", expansion_id);
+		break;
 	}
 
 	if (expansion_config.content == 1)
diff --git a/board/overo/overo.h b/board/overo/overo.h
index b984a54..89679d6 100644
--- a/board/overo/overo.h
+++ b/board/overo/overo.h
@@ -420,4 +420,12 @@ const omap3_sysinfo sysinfo = {
 	MUX_VAL(CP(MCSPI1_SOMI),	(IEN  | PTD | DIS | M4)) /*GPIO_173 */\
 	MUX_VAL(CP(MCSPI1_CS1),		(IDIS | PTD | EN  | M4)) /*GPIO_175 */\
 
+#define MUX_ALTO35() \
+	MUX_VAL(CP(SYS_CLKOUT1),	(IEN  | PTU | EN  | M4)) /*GPIO_10-BTN*/\
+	MUX_VAL(CP(UART1_TX),		(IDIS | PTD | DIS | M4)) /*GPIO_148-RED LED*/\
+	MUX_VAL(CP(UART1_CTS),		(IDIS | PTD | DIS | M4)) /*GPIO_150-YELLOW LED*/\
+	MUX_VAL(CP(UART1_RX),		(IDIS | PTD | DIS | M4)) /*GPIO_151-BLUE LED*/\
+	MUX_VAL(CP(HDQ_SIO),		(IDIS | PTD | DIS | M4)) /*GPIO_170-GREEN LED*/\
+	MUX_VAL(CP(MCSPI1_CS1),		(IDIS | PTD | EN  | M4)) /*GPIO_175*/\
+
 #endif
-- 
1.7.9.5

